components/infobox/wikis/clashofclans/infobox_building_custom.lua

```diff
@@ -0,0 +1,87 @@
+---
+-- @Liquipedia
+-- wiki=clashofclans
+-- page=Module:Infobox/Building/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local Building = Lua.import('Module:Infobox/Building', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+
+---@class CustomBuildingInfobox: BuildingInfobox
+local CustomBuilding = Class.new()
+
+---@class CustomWidgetInjector: WidgetInjector
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _pagename = mw.title.getCurrentTitle().text
+
+---@param frame Frame
+---@return Html
+function CustomBuilding.run(frame)
+	local building = Building(frame)
+	_args = building.args
+	_args.informationType = 'Building'
+
+	building.getWikiCategories = CustomBuilding.getWikiCategories
+	building.setLpdbData = CustomBuilding.setLpdbData
+	building.createWidgetInjector = CustomBuilding.createWidgetInjector
+
+	return building:createInfobox()
+end
+
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:addCustomCells(widgets)
+	Array.appendWith(
+		widgets,
+		Cell{name = 'Range', content = {_args.range}},
+		Cell{name = 'Damage Type', content = {_args.damagetypo}},
```

Comment:
you sure the param is correct?
shouldn't it be `damagetype` (e at the end not o)